ste/downloader-blob.go

```diff
@@ -66,9 +66,23 @@ func newBlobDownloader(jptm IJobPartTransferMgr) (downloader, error) {
 		return nil, err
 	}
 
+	blobClient := s.NewContainerClient(jptm.Info().SrcContainer).NewBlobClient(jptm.Info().SrcFilePath)
+
+	if jptm.Info().VersionID != "" {
+		blobClient, err = s.NewContainerClient(jptm.Info().SrcContainer).NewBlobClient(jptm.Info().SrcFilePath).WithVersionID(jptm.Info().VersionID)
```

Comment:
there's no need to re-do the call to NewContainerClient and NewBlobClient here since we do that above. We can just take the blobClient above and call WithVersionId or WithSnapshot

Reply:
resolved in new pr https://github.com/Azure/azure-storage-azcopy/pull/2527